pkg/cli/dev.go

```diff
@@ -0,0 +1,180 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	v1 "github.com/acorn-io/acorn/pkg/apis/api.acorn.io/v1"
+	cli "github.com/acorn-io/acorn/pkg/cli/builder"
+	"github.com/acorn-io/acorn/pkg/client"
+	"github.com/acorn-io/acorn/pkg/deployargs"
+	"github.com/acorn-io/acorn/pkg/dev"
+	"github.com/acorn-io/acorn/pkg/rulerequest"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+func NewDev(c CommandContext) *cobra.Command {
+	cmd := cli.Command(&Dev{out: c.StdOut, client: c.ClientFactory}, cobra.Command{
+		Use:               "dev [flags] IMAGE|DIRECTORY [acorn args]",
+		SilenceUsage:      true,
+		Short:             "Run an app from an image or Acornfile in dev mode or attach a dev session to a currently running app",
+		ValidArgsFunction: newCompletion(c.ClientFactory, imagesCompletion(true)).withSuccessDirective(cobra.ShellCompDirectiveDefault).withShouldCompleteOptions(onlyNumArgs(1)).complete,
+		Example: `
+acorn dev <IMAGE>
+acorn dev .
+acorn dev --name wandering-sound
+acorn dev --name wandering-sound <IMAGE>
+`})
+
+	// This will produce an error if the volume flag doesn't exist or a completion function has already
+	// been registered for this flag. Not returning the error since neither of these is likely occur.
+	if err := cmd.RegisterFlagCompletionFunc("volume", newCompletion(c.ClientFactory, volumeFlagClassCompletion).complete); err != nil {
+		cmd.Printf("Error registering completion function for -v flag: %v\n", err)
+	}
+	// This will produce an error if the computeclass flag doesn't exist or a completion function has already
+	// been registered for this flag. Not returning the error since neither of these is likely occur.
+	if err := cmd.RegisterFlagCompletionFunc("compute-class", newCompletion(c.ClientFactory, computeClassFlagCompletion).complete); err != nil {
+		cmd.Printf("Error registering completion function for --compute-class flag: %v\n", err)
+	}
+	cmd.PersistentFlags().Lookup("dangerous").Hidden = true
+	cmd.Flags().SetInterspersed(false)
+	return cmd
+}
+
+type Dev struct {
+	RunArgs
+	BidirectionalSync bool `usage:"In interactive mode download changes in addition to uploading" short:"b"`
+	Quiet             bool `usage:"Do not print status" short:"q"`
+	out               io.Writer
+	client            ClientFactory
+}
+
+func (s *Dev) Run(cmd *cobra.Command, args []string) error {
+	c, err := s.client.CreateDefault()
+	if err != nil {
+		return err
+	}
+
+	// Force install prompt if needed
+	_, err = c.Info(cmd.Context())
+	if err != nil {
+		return err
+	}
+	cwd := "."
+	if len(args) > 0 {
+		cwd = args[0]
+	}
+	//acorn args set cwd back to .
+	if strings.HasPrefix(cwd, "-") {
+		cwd = "."
+	}
+
+	image := cwd
+	isDir, err := isDirectory(cwd)
```

Comment:
I think so, but it a lot of the logic is for determining if it is an image vs running app vs dir... but it could probably be simplified if we know for sure it can only be "."